idrstream/idr_stream.py

```diff
@@ -85,9 +99,17 @@ def __init__(
         """
         self.idr_id = idr_id
         self.tmp_dir = tmp_dir
-        # DP project will be in tmp dir
-        self.DP_project_path = pathlib.Path(f"{tmp_dir}/DP_project/")
-        self.DP_project_path.mkdir(parents=True, exist_ok=True)
+
+        if stream_type == "DP":
+            # DP project will be in tmp dir
+            self.DP_project_path = pathlib.Path(f"{tmp_dir}/DP_project/")
+            self.DP_project_path.mkdir(parents=True, exist_ok=True)
+
+        if stream_type == "CP":
+            # CP project will be in tmp dir
+            self.CP_project_path = pathlib.Path(f"{tmp_dir}/CP_project/")
+            self.CP_project_path.mkdir(parents=True, exist_ok=True)
```

Comment:
@d33bs 

I am not quite sure what you mean here. Are you suggesting to add a `project_path` parameter instead? Thank you in advance for clarification :smile:

Reply:
Thank you for asking about this. There are a number of spots with class attributes and methods in this file which use `CP_` or `DP_` as part of the name, which may be cause for a greater number of lines of code as a result. This could be cause for greater maintenance and decreased readability.

In some spots, like this one, there seems like an opportunity to use `stream_type` as a way to distinguish those attributes and methods in an abstracted way (as you suggested, with `project_path`). `stream_type` might also be useful as a class attribute (`self.stream_type`), depending on how it's used by methods throughout the class. 

Worth mentioning: I may be incorrect in calling this topic out, please use your best judgement. There are occasions where the sacrifice of a few extra lines for specificity is of greater value for readability and simplicity (or decreasing the risk that something is used inappropriately). It's a delicate balance between [lines 25-27 of PEP 20](https://github.com/python/peps/blob/main/pep-0020.txt#L25)!

Reply:
Fixed and added `stream_type` as a class attribute :smile: